eng/scripts/MavenPackaging.ps1

```diff
@@ -153,3 +153,66 @@ function Get-FilteredMavenPackageDetails([string]$ArtifactDirectory, [string]$Gr
 
   return $filteredPackageDetails
 }
+
+# Compare the contents of SHA hash files for a local repository to the same
+# files in Maven Central or Azure DevOps. Returns $false if none of the hash
+# files exist in the remote, $true if all hashes exist and match the local
+# hash, otherwise throws.
+function Test-ReleasedPackage([string]$RepositoryUrl, [MavenPackageDetail]$PackageDetail, [string]$BearerToken) {
+  if ($RepositoryUrl -match "^https://pkgs.dev.azure.com/azure-sdk/\b(internal|public)\b/*") {
+    if(!$BearerToken) {
+      throw "BearerToken required for Azure DevOps package feeds"
+    }
+    
+    $baseUrl = "https://pkgs.dev.azure.com/azure-sdk/internal/_packaging/azure-sdk-for-java-pr/maven/v1"
+    $algorithm = "sha256"
+    $headers = @{ Authorization="BEARER $BearerToken" }
+  }
+  elseif ($RepositoryUrl -match "^https://oss.sonatype.org/service/local/staging/deploy/maven2") {
+    $baseUrl = "https://repo1.maven.org/maven2"
+    $algorithm = "sha1"
+  }
+  else {
+    throw "Repository URL must be either an Azure Artifacts feed, or a SonaType Nexus feed."
+  }
+
+  $packageUrl = "$baseUrl/$($PackageDetail.GroupId.Replace('.', '/'))/$($PackageDetail.ArtifactID)/$($PackageDetail.Version)"
+  
+  # Count the number of remote hashes found
+  $remoteCount = 0
+
+  # Count the number of remote hashes that match their local hash
+  $matchCount = 0
+
+  foreach($artifact in $PackageDetail.AssociatedArtifacts) {
+    $localFileName = $artifact.File.Name;
+    $remoteHashUrl = "$packageUrl/$localFileName.$algorithm"
+
+    Write-Information "Comparing local and remote hashes for $localFileName"
+    Write-Information "  Getting remote hash"
+    $response = Invoke-WebRequest -Method GET -Uri $remoteHashUrl -Headers $headers -SkipHttpErrorCheck
+
+    if($response.StatusCode -ge 200 -and $response.StatusCode -lt 300) {
+      $remoteCount++
+      $remoteHash = $response.Content
+
+      Write-Information "  Getting local hash"
+      $localPath = $artifact.File.FullName
+      $localHash = Get-FileHash -Path $localPath -Algorithm $algorithm | Select-Object -ExpandProperty 'Hash'
```

Comment:
Do we want to compute the has? Should we already have the hash file present locally?

Reply:
We don't have sha256 or 512 hashes local for the Azure DevOps releases.  It's trivial to generate them and keeps the Sonotype/ADO code symmetric.

Reply:
OK as long as we are computing the hash the same way it shouldn't matter.

Do you know where the sha256 files are coming from for DevOps? Does devops compute them on upload?

Reply:
I can't tell.  I've been unable to Fiddler the requests.